### PR TITLE
[connection] update path challenge according to the draft [JF & CC]

### PR DESCRIPTION
**Pull request by ElNiak**
> " Path validation succeeds when a PATH_RESPONSE frame is received that
>    contains the data that was sent in a previous PATH_CHALLENGE frame.
>    A PATH_RESPONSE frame received on any network path validates the path
>    on which the PATH_CHALLENGE was sent."

We try to modify the PATH_CHALLENGE part of the connection.py to follow this rule that we found with a testing tool Ivy. The problem before was that if the PATH_RESPONSE was received on a different path the challenge was considered as failed. 

JF & CC

### CODE REVIEW

**Comment by jlaine** [comment]
This looks like a good catch. Would you mind cleaning up the patch (removing commented code and debug statements)?

**Comment by jlaine** [comment]
Hello @ElNiak any chance of getting the code into a shape I can merge it?

**Comment by ElNiak** [comment]
Hello, sorry we forget about that it will be done asap !

**Comment by jlaine** [comment]
The PR seems to completely break the test suite would you mind updating it?

File: src/aioquic/quic/connection.py [hunk 1]
```diff
@@ -1147,6 +1162,17 @@ def _find_network_path(self, addr: NetworkAddress) -> QuicNetworkPath:
         self._logger.debug("Network path %s discovered", network_path.addr)
         return network_path
 
+    def _find_network_challenge(self) -> QuicNetworkPathChallenge:
+        # check existing network challenge
+        for idx, network_chall in enumerate(self._network_challenges):
```
Comment (jlaine) [inline src/aioquic/quic/connection.py:1167]: This looks like a complicated way of saying "always take the last challenge in the list"?

**Comment by jlaine** [comment]
Hi @ElNiak do you plan to complete the work on this PR?

**Comment by rthalley** [comment]
Closing this in favor of the approach of [#483].

**Pull request closed without merging**